scarlet/operator.py

```diff
@@ -259,24 +309,55 @@ def prox_uncentered_symmetry(X, step, center=None, strength=.5, use_soft=True, f
         The parameter to update.
     step: `int`
         Step size of the gradient step.
-    center: tuple
+    center: tuple of `int`
         The center pixel coordinates to apply the symmetry operator.
+    algorithm: `string`
+        The algorithm to use for symmetry.
+        * If `algorithm = "kspace" then `X` is shifted by `shift` and
+          symmetry is performed in kspace. This is the only ymmetry algorithm
+          in scarlet that works for fractional pixel shifts.
+        * If `algorithm = "sdss" then the SDSS symmetry is used,
+          namely the source is made symmetric around the `center` pixel
+          by taking the minimum of each pixel and its symmetric partner.
+          This is the algorithm used when initializing an `ExtendedSource`
+          because it keeps the morphologies small, but during optimization
+          the penalty is much stronger than the gradient
+          and often leads to vanishing sources.
+        * If `algorithm = "soft" then soft symmetry is used,
```

Comment:
How useful is `soft` now? 
Do we still have a need for treating perfectly centered sources differently?

Reply:
It is faster to to the direct symmetry, since it's just `(x + x[::-1, ::-1])/2`, so we should use that if there isn't a shift. I don't necessarily see the need for the `strength` parameter but there doesn't seem to be any harm in keeping it.